ArduCopter/mode_follow.cpp

```diff
@@ -106,8 +102,8 @@ void ModeFollow::run()
         const float des_vel_z_max = copter.avoid.get_max_speed(pos_control->get_pos_z_p().kP().get(), pos_control->get_max_accel_z() * 0.5f, fabsf(dist_vec_offs_neu.z), copter.G_Dt);
         desired_velocity_neu_cms.z = constrain_float(desired_velocity_neu_cms.z, -des_vel_z_max, des_vel_z_max);
 
-        // get avoidance adjusted climb rate
-        desired_velocity_neu_cms.z = get_avoidance_adjusted_climbrate(desired_velocity_neu_cms.z);
+        // limit the velocity for obstacle/fence avoidance
+        copter.avoid.adjust_velocity(pos_control->get_pos_xy_p().kP().get(), pos_control->get_max_accel_xy(), desired_velocity_neu_cms, pos_control->get_pos_z_p().kP().get(), pos_control->get_max_accel_z(), G_Dt);
```

Comment:
It might be better to move the desired velocity vector to be the first argument.  I think this is a more standard way to order argument rather than putting it in the middle.